diff --git a/CMakeLists.txt b/CMakeLists.txt
index 14568ee0aa8..0d067691903 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -844,6 +844,9 @@ if( MINGW AND NOT "${CMAKE_CXX_COMPILER_ID}" MATCHES "Clang" )
   llvm_replace_compiler_option(CMAKE_CXX_FLAGS_RELEASE "-O3" "-O2")
 endif()

+# Ubuntu 22 moved numeric_limits into limits (out of std)
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -include limits")
+
 # Put this before tblgen. Else we have a circular dependence.
 add_subdirectory(lib/Demangle)
 add_subdirectory(lib/Support)
