diff --git a/tools/external/wabt/src/lexer-source.cc b/tools/external/wabt/src/lexer-source.cc
index 762aea7..1200b58 100644
--- a/tools/external/wabt/src/lexer-source.cc
+++ b/tools/external/wabt/src/lexer-source.cc
@@ -38,7 +38,8 @@ std::unique_ptr<LexerSource> LexerSourceFile::Clone() {
     result.reset();
   }
 
-  return std::move(result);
+  //return std::move(result);
+  return result;
 }
 
 Result LexerSourceFile::Tell(Offset* out_offset) {
